hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddedForeignKeyDescriptor.java

```diff
@@ -181,11 +186,11 @@ public DomainResult createDomainResult(
 							),
 							s ->
 									new ColumnReference(
-											identificationVariable,
```

Comment:
I'm tempted to say that `ColumnReference` should accept the `ColumnMapping` rather than the individual pieces.  I understand you probably chose that because it matches the existing contracts.  

But this is point I brought up in regards to model parts accepting a `ColumnConsumer`, but that really a `ColumnMappingConsumer` makes sense as well.  There is never a time when a model part wont have the  `ColumnMapping` reference(s) already, so concerns about instantiations isn't really a hinderance there.

I'm not sure there is ever a time when we would not want a `ColumnMapping` on the `ColumnReference` for some reason ("PK tables", e.g.), but if so we could e.g. add a `ModelColumnReference` also that takes the ColumnMapping.

I just think we added this `ColumnMapping` contract because it cleaned up a lot of code.  It would be a shame to not use it in other places where we could be using it.